python-package/setup.py

```diff
@@ -251,6 +258,66 @@ def run(self):
             os.remove(LOG_PATH)
 
 
+class CustomBdistWheel(bdist_wheel):
+
+    user_options = bdist_wheel.user_options + [
+        ('mingw', 'm', 'Compile with MinGW'),
+        ('integrated-opencl', None, 'Compile integrated OpenCL version'),
+        ('gpu', 'g', 'Compile GPU version'),
+        ('cuda', None, 'Compile CUDA version'),
+        ('mpi', None, 'Compile MPI version'),
+        ('nomp', None, 'Compile version without OpenMP support'),
+        ('hdfs', 'h', 'Compile HDFS version'),
+        ('bit32', None, 'Compile 32-bit version'),
+        ('precompile', 'p', 'Use precompiled library'),
+        ('boost-root=', None, 'Boost preferred installation prefix'),
+        ('boost-dir=', None, 'Directory with Boost package configuration file'),
+        ('boost-include-dir=', None, 'Directory containing Boost headers'),
+        ('boost-librarydir=', None, 'Preferred Boost library directory'),
+        ('opencl-include-dir=', None, 'OpenCL include directory'),
+        ('opencl-library=', None, 'Path to OpenCL library')
+    ]
```

Comment:
Please deduplicate code by creating global variable `LIGHTGBM_OPTIONS` in `setup.py` and using it here and in `CustomInstall` class
https://github.com/microsoft/LightGBM/blob/b360700411a31085af2319294a5553e7c667a242/python-package/setup.py#L197-L213